src/pseudopeople/interface.py

```diff
@@ -52,7 +52,10 @@ def _generate_dataset(
             f"No datasets found at directory {str(source)}. "
             "Please provide the path to the unmodified root data directory."
         )
-    suffix = set(x.suffix for x in data_paths)
+    if isinstance(data_paths, dict):
+        suffix = set(x.suffix for item in list(data_paths.values()) for x in item)
+    else:
+        suffix = set(x.suffix for x in data_paths)
```

Comment:
Is there a cleaner way to do this?

Reply:
Could be cleaner to extract everything from line 55 to line 63 into a method called `validate_data_path_suffix()`